sorc/ncep_post.fd/SURFCE.f

```diff
@@ -848,8 +858,12 @@ SUBROUTINE SURFCE
         if(grib=='grib2') then
           cfld=cfld+1
           fld_info(cfld)%ifld=IAVBLFLD(IGET(502))
-          fld_info(cfld)%ntrange=IFHR-ID(18)
-          fld_info(cfld)%tinvstat=1
+            if(ITSRFC>0) then
```

Comment:
@EricJames-NOAA "ITSRFC" is defined in the section of writing ID 500 only. It would be better to add a line "ITSRFC     = NINT(TSRFC)" in all sections with your bug fix (such as ID 501, 502, 503, 504, 506, 507).